### PR TITLE
Add slug regex checks

### LINKED ISSUE #100: [3] Add slug validating regex
Valid:
            test
            a1b2-c3

          Invalid:
            1a2b
            a1b2--c3

### PR DESCRIPTION
**Pull request by tschuy**
This fixes #100 .

There are comments in the the regex for what it SHOULD match. If you think of a test case that should be added, please mention it.

### CODE REVIEW

File: tests/test.js [hunk 1]
```diff
@@ -59,7 +59,6 @@ describe('Endpoints', function() {
     require('./times')(expect, request, baseUrl);
     require('./activities')(expect, request, baseUrl);
     require('./projects')(expect, request, baseUrl);
-    require('./helpers')(expect);
```
Comment (Kennric) [inline tests/test.js]: is this intentionally removed?

Comment (tschuy) [inline tests/test.js]: Yes, it's duplicated under both Endpoints and Helpers. It's not an endpoint :P

Comment (Kennric) [inline tests/test.js]: Ah, ok, I was thinking this was responsible for making the helpers tests run.

**Comment by Kennric** [comment]
This looks good to me, regex seems reasonable thorough, and I think we have a good standard for a slug - the one question I have is, should we exclude leading integers? I am not sure if that is useful or important.

**Comment by iankronquist** [comment]
What's the spec for slugs? I thought it was [the same as Django's](https://github.com/osuosl/timesync-node/blob/develop/docs/source/design/draft_models.rst#draft-data-models). If so, we should copy [Django's slug regex](https://github.com/django/django/blob/master/django/core/validators.py#L211) instead of rolling our own which matches different things.

**Comment by tschuy** [comment]
@iankronquist it's https://github.com/osuosl/timesync/pull/16

**Comment by ElijahCaine** [comment]
I tested the regex  (by hand). The tests pass and seem to cover all of our bases. The fact that our slug definition is a little wonky isn't ideal, but the rules are simple enough (and in the comments!). +1

**Comment by ElijahCaine** [comment]
We should consider changing the regex from `^[a-z]+[a-z0-9]*(?:-[a-z0-9]+)*$``` to ``^(?:[0-9]*-)*[0-9]*[a-z]+[a-z0-9]*(?:-[a-z0-9]+)*$`` as it allows us to have preceding numbers but does not allow for the regex to just be a number.

**Comment by ElijahCaine** [comment]
This new regex structure is nice and easy to read. The tests pass and they seem to cover all of our bases. Much nicer than the alternative of one giant regex monstrosity. +1

**Comment by jordane** [comment]
@tschuy There are no tests for the regex that contain non-alphanumeric characters. It'd be wise to add those in case the regexes get refactored later.

**Comment by ElijahCaine** [comment]
Right. I completely didn't catch that it was missing non-alphanumeric tests. These look good. Hopefully 15 tests for the regex is good enough. I can't think of any more input that we could test for. +1 (again)

**Comment by ElijahCaine** [comment]
Everything *still* looks good. Post +1 from me.

**Comment by Kennric** [comment]
This looks pretty thorough, I can't help feeling that our regex could be simpler, but it works, the tests test our specified slug criteria, and the regex can be streamlined later if we figure out how. +1

**Pull request merged**